src/components/Admin/AddQuestion.js

```diff
@@ -0,0 +1,259 @@
+            console.log("Upload is " + progress + "% done");
+          },
+          function (error) {
+            console.log(error);
+          },
+          function () {
+            console.log("Upload Successful!");
+          }
+        );
+      }
+
+      this.props.firebase
+        .questions()
+        .add(data)
+        .then(() => {
+          console.log("Added Question");
+          this.setState(
+            {
+              submitted: true,
+            },
+            () => {
+              this.props.updateFunc();
+            }
+          );
+        })
+        .catch((error) => {
+          this.setState({ error });
+        });
+    }
+
+    this.setState({ sending: false });
+    event.preventDefault();
+  };
+
+  onChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  onFileChange = (event) => {
+    var f = event.target.files[0];
+    console.log(f);
+    if (f.type !== "image/jpeg" && f.type !== "image/png") {
+      console.log("Invalid file type");
+      f = null;
+    } else {
+      var fileExtension = "jpg";
+      if (f.type.split("/")[1] === "png") fileExtension = "png";
+      this.setState({ fileExtension: fileExtension });
+    }
+
+    this.setState({ file: f });
+  };
+
+  render() {
+    const {
+      fileExtension,
+      file,
+      name,
+      answer,
+      description,
+      loading,
+      error,
+      submitted,
+      validated,
+      sending,
+    } = this.state;
+
+    if (loading) return <Loader />;
+
+    if (error)
+      return (
+        <Container flexdirection="column">
+          <h1>Uh oh!</h1>
+          <p>{error}</p>
+        </Container>
+      );
+
+    return (
+      <StyledContainer flexdirection="column">
+        <CenteredForm noValidate validated={validated} onSubmit={this.onSubmit}>
+          <h2> Add Question </h2>
+          <Form.Row style={{ width: "100%" }} key={0}>
+            <Form.Group controlId={4} style={{ width: "100%" }}>
+              <Form.Label>
+                <h5> Name </h5>
+              </Form.Label>
+              <Form.Control
+                name="name"
+                type="text"
+                placeholder="..."
+                value={name}
+                onChange={this.onChange}
+              />
+            </Form.Group>
+          </Form.Row>
+
+          <Form.Row style={{ width: "100%" }} key={1}>
+            <Form.Group controlId={4} style={{ width: "100%" }}>
+              <Form.Label>
+                <h5> Answer </h5>
+              </Form.Label>
+              <Form.Control
+                rows={5}
+                name="answer"
+                type="textarea"
+                placeholder="..."
+                value={answer}
+                onChange={this.onChange}
+              />
+            </Form.Group>
+          </Form.Row>
+
+          <Form.Row style={{ width: "100%" }} key={2}>
+            <Form.Group controlId={4} style={{ width: "100%" }}>
```

Comment:
If there's a style you're repeating that's a good opportunity to use a styled component

Reply:
Resolved in commit 328335366bf582d828349cde311f9f15794bbca6